pkg/applicationserver/applicationserver.go

```diff
@@ -1026,7 +1026,7 @@ func (as *ApplicationServer) decryptDownlinkMessage(ctx context.Context, ids ttn
 	case dev.PendingSession != nil && bytes.Equal(dev.PendingSession.SessionKeyID, msg.SessionKeyID):
 		session = dev.PendingSession
 	}
-	if session.GetAppSKey() == nil {
+	if session == nil || session.GetAppSKey() == nil {
```

Comment:
Would it be better to add a default to the `Switch` and exit the function with a `errSessionNotFound`?